Enhancement/4052 trigger remote survey

eugene-manuilov opened the pull request:
## Summary

<!-- Please reference the issue this PR addresses. -->
Addresses issue #4052

## Relevant technical choices

* Decided not to create a new hook because there is no sense in it since we would use it only for one component.

## Checklist

- [ ] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [ ] I have added a QA Brief on the issue linked above.
- [ ] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

aaemnnosttv commented on assets/js/modules/idea-hub/datastore/module-data.js:
```diff
@@ -65,18 +66,48 @@ export const actions = {
+		case INCREMENT_INTERACTION_COUNT: {
+			const { ideaHubData = {} } = state;
+
+			let interactionCount = parseInt( ideaHubData.interactionCount );
```
```suggestion
			let interactionCount = parseInt( ideaHubData.interactionCount, 10 );
```

aaemnnosttv commented on assets/js/modules/idea-hub/datastore/module-data.js:
```diff
@@ -65,18 +66,48 @@ export const actions = {
+			const { ideaHubData = {} } = state;
+
+			let interactionCount = parseInt( ideaHubData.interactionCount );
+			if ( Number.isNaN( interactionCount ) ) {
+				interactionCount = 0;
+			}
```
We could use `0` as a fallback instead, e.g. `parseInt(...) || 0`?

aaemnnosttv commented on assets/js/modules/idea-hub/datastore/module-data.test.js:
```diff
@@ -0,0 +1,134 @@
+import { MODULES_IDEA_HUB } from './constants';
+
+describe( 'modules/idea-hub module-data', () => {
+	const baseInfoVar = '_googlesitekitIdeaHub';
+	const baseInfo = {
```
These look like they were copied from the `core/site` tests. Let's rename these to be more specific to this test.
```suggestion
	const ideaHubDataGlobal = '_googlesitekitIdeaHub';
```
As for `baseInfo` here, it looks like it's only used in the one test for selectors, so how about we inline it there instead?

aaemnnosttv commented on includes/Modules/Idea_Hub.php:
```diff
@@ -106,20 +107,23 @@ final class Idea_Hub extends Module
 	/**
 	 * Post_Idea_Name instance.
 	 *
+	 * @since n.e.x.t
 	 * @var Post_Idea_Name
 	 */
 	private $post_name_setting;
 
 	/**
 	 * Post_Idea_Text instance.
 	 *
+	 * @since n.e.x.t
 	 * @var Post_Idea_Text
 	 */
 	private $post_text_setting;
 
 	/**
 	 * Post_Idea_Topics instance.
 	 *
+	 * @since n.e.x.t
```
Nice catch! Let's update these to use the correct version though as `n.e.x.t` would not be accurate.

aaemnnosttv commented on assets/js/modules/idea-hub/components/dashboard/DashboardIdeasWidget/index.js:
```diff
@@ -82,15 +84,19 @@ function DashboardIdeasWidget( props ) {
+			newIdeas: select( MODULES_IDEA_HUB ).getNewIdeas(),
+			savedIdeas: select( MODULES_IDEA_HUB ).getSavedIdeas(),
+			draftIdeas: select( MODULES_IDEA_HUB ).getDraftPostIdeas(),
+			interactionsCount: select( MODULES_IDEA_HUB ).getInteractionCount(),
```
Let's make this singular like it is in the selector (and PHP).

```suggestion
			interactionCount: select( MODULES_IDEA_HUB ).getInteractionCount(),
```

aaemnnosttv commented on assets/js/modules/idea-hub/components/dashboard/DashboardIdeasWidget/index.js:
```diff
@@ -82,15 +84,19 @@ function DashboardIdeasWidget( props ) {
-		select( MODULES_IDEA_HUB ).getNewIdeas()
+	const [ triggeredSurvey, setTriggeredSurvey ] = useState( false );
+
+	const { newIdeas, savedIdeas, draftIdeas, interactionsCount } = useSelect(
```
This also pending the result of our call later.

aaemnnosttv commented on assets/js/components/ModulesList.js:
```diff
@@ -110,5 +110,5 @@ export default function ModulesList( { moduleSlugs } ) {
 
 ModulesList.propTypes = {
-	moduleSlug: PropTypes.arrayOf( PropTypes.string ).isRequired,
+	moduleSlugs: PropTypes.arrayOf( PropTypes.string ).isRequired,
```
This just fixes an unrelated prop type error, correct?

aaemnnosttv reviewed (commented):
Thanks @eugene-manuilov ! This looks solid to me, just a few minor things to touch up and then this should be good to go 👍

felixarntz reviewed (changes requested):
@eugene-manuilov I found one problem with the implementation while testing.

Regarding your question: In order to get a survey to be returned, you need to technically be on Site Kit version 1.42.0 or higher. In order to do that (and have the proxy recognize it), you'll need to e.g. manually update the version numbers in your Site Kit codebase, and then either wait for the refresh token call to trigger or just disconnect/reset and reconnect. Any of those will update the proxy data with the current Site Kit version used on the site.

eugene-manuilov commented:
@felixarntz @aaemnnosttv going to close this PR because it contains unwanted changes from the develop branch. Please, take a look at #4100. It contains all changes and addresses code review feedback added for this PR.
